src/huggingface_hub/inference/_common.py

```diff
@@ -197,6 +198,18 @@ def _b64_encode(content: ContentT) -> str:
         return base64.b64encode(data_as_bytes).decode()
 
 
+def _as_url(content: ContentT, default_mime_type: str) -> str:
+    if isinstance(content, str) and (content.startswith("https://") or content.startswith("http://")):
+        return content
+
+    mime_type: Optional[str] = None
+    if isinstance(content, (str, Path)):
+        mime_type, _ = mimetypes.guess_type(str(content))
```

Comment:
```suggestion
        mime_type, _ = mimetypes.guess_type(content, strict=False)
```

from [official docs](https://docs.python.org/3.10/library/mimetypes.html#mimetypes.guess_type) (I didn't know about this module^^):
- _URL can be a string or a [path-like object](https://docs.python.org/3.10/glossary.html#term-path-like-object)_
=> so no need to convert to `str`
-  _When strict is True (the default), only the IANA types are supported; when strict is False, some additional non-standard but commonly used MIME types are also recognized._
=> let's not be strict, if the module can guess a non-standard mimetype then let's still use it